src/main/java/impl/image/ImageConverterImpl.java

```diff
@@ -0,0 +1,171 @@
+                        return constructColor(value);
+                    }
+                };
+            } catch (NoSuchFieldException e) {
+                colorValueSetter = this::constructColor;
+            }
+        }
+        return colorValueSetter;
+    }
+
+
+    /**
+     * Transforms 2d array of type S to 2d array of type T
+     * @param <S> Source type
+     * @param <T> Target type
+     */
+    private static class ArrayTransformer<S, T> {
+
+        /**
+         * Transforms 2d array of type S to 2d array of type T using given converter function that is applied element-wise
+         * @param source source 2d array of type S
+         * @param converter converter function that is applied to each element
+         * @param targetClass Class object of target type, needed to be able to access target ctor cause of type erasure.
+         * @return converted 2d array of type T
+         * @implNote probably not the best looking impl since I can imagine same s possible using streams but
+         * I couldn't get that right...
+         */
+        @SuppressWarnings("unchecked")
+        public T[][] transform(S[][] source, Function<S, T> converter, Class<T> targetClass) {
+//            T[][] result = new T[source.length][]; //oh, yeah, type erasure, I can't do this...
+//            Ok, there's sorta workaround we can pass target class obj...
+//            And since we need nested Arrays... and Class<T[]> is not a thing.....
+            Class targetArrayClass = Array.newInstance(targetClass, 0).getClass();
+            T[][] result = (T[][]) Array.newInstance(targetArrayClass, source.length);
+            int rowIndex = 0;
+            for (var row : source) {
+                result[rowIndex] = (T[]) Array.newInstance(targetClass, row.length);
+                int colIndex = 0;
+                for (var item : row) {
+                    result[rowIndex][colIndex] = converter.apply(item);
+                    colIndex += 1;
+                }
+                rowIndex += 1;
+            }
+            return result;
+        }
+    }
+
+    private Integer[][] box2dArray(int[][] array) {
+        return Arrays.stream(array).map(
+                (line) -> Arrays.stream(line).boxed().toArray(Integer[]::new)
+        ).toArray(Integer[][]::new);
+    }
+
+
+    private int[][] unbox2dArray(Integer[][] array) {
+        return Arrays.stream(array).map(
+                line -> Arrays.stream(line).mapToInt(Integer::intValue).toArray()
+        ).toArray(int[][]::new);
+    }
+
+    @Override
+    public Color[][] convertToColor(int[][] image) {
+//        return new ArrayTransformer<Integer, Color>().transform(image, this::constructColor, Color.class);
+//        DAMN, primitives can't be generic params! second bump =(
+//        How do I squeeze damn ints into all the generic thing without second n^2 iteration...
+//        screw this, I'm doing copy-paste... OR double iterating for boxing-unboxing... how do you...
```

Comment:
It is very commendable that you generalized the task. But one must understand whether it is worth doing generalization in such a situation

Reply:
I said I was sorry for this =) I just got a little bored and wanted to have some fun